### ISSUE TITLE
Rails 7 installation fails (undefined method action_mailer) when some components are excluded

### ISSUE THREAD

**Issue by ekortright** [rails/rails#47206]
### Steps to reproduce
Creating a new Rails app inside a Docker container (that's why it's running as root), excluding components that are not needed.

```ruby
bash-4.2# rails new --skip-action-mailbox --skip-action-text --skip-active-record --skip-active-job --skip-active-storage --skip-action-cable --skip-asset-pipeline --skip-hotwire myApp
```

### Expected behavior
Creation task should completely execute without error.

### Actual behavior
The task is interrupted before it can finish because it is expecting `action_mailer` to be a method in Rails::Application::Configuration.  The problem seems to be that `config/application.rb` does not require all gems, but has a separate require for each gem, and the one for `action_mailer` is commented out:

```
# require "action_mailer/railtie"
```
Here is the output of the `rails new` command:

```
      create  
      create  README.md
      create  Rakefile
      create  .ruby-version
      create  config.ru
      create  .gitignore
      create  .gitattributes
      create  Gemfile
         run  git init from "."
Initialized empty Git repository in /rails/myApp/.git/
      create  app
      create  app/assets/config/manifest.js
      create  app/assets/stylesheets/application.css
      create  app/channels/application_cable/channel.rb
      create  app/channels/application_cable/connection.rb
      create  app/controllers/application_controller.rb
      create  app/helpers/application_helper.rb
      create  app/jobs/application_job.rb
      create  app/mailers/application_mailer.rb
      create  app/models/application_record.rb
      create  app/views/layouts/application.html.erb
      create  app/views/layouts/mailer.html.erb
      create  app/views/layouts/mailer.text.erb
      create  app/assets/images
      create  app/assets/images/.keep
      create  app/controllers/concerns/.keep
      create  app/models/concerns/.keep
      create  bin
      create  bin/rails
      create  bin/rake
      create  bin/setup
      create  config
      create  config/routes.rb
      create  config/application.rb
      create  config/environment.rb
      create  config/puma.rb
      create  config/environments
      create  config/environments/development.rb
      create  config/environments/production.rb
      create  config/environments/test.rb
      create  config/initializers
      create  config/initializers/assets.rb
      create  config/initializers/content_security_policy.rb
      create  config/initializers/cors.rb
      create  config/initializers/filter_parameter_logging.rb
      create  config/initializers/inflections.rb
      create  config/initializers/new_framework_defaults_7_0.rb
      create  config/initializers/permissions_policy.rb
      create  config/locales
      create  config/locales/en.yml
      create  config/master.key
      append  .gitignore
      create  config/boot.rb
      create  lib
      create  lib/tasks
      create  lib/tasks/.keep
      create  lib/assets
      create  lib/assets/.keep
      create  log
      create  log/.keep
      create  public
      create  public/404.html
      create  public/422.html
      create  public/500.html
      create  public/apple-touch-icon-precomposed.png
      create  public/apple-touch-icon.png
      create  public/favicon.ico
      create  public/robots.txt
      create  tmp
      create  tmp/.keep
      create  tmp/pids
      create  tmp/pids/.keep
      create  tmp/cache
      create  tmp/cache/assets
      create  vendor
      create  vendor/.keep
      create  test/fixtures/files
      create  test/fixtures/files/.keep
      create  test/controllers
      create  test/controllers/.keep
      create  test/mailers
      create  test/mailers/.keep
      create  test/models
      create  test/models/.keep
      create  test/helpers
      create  test/helpers/.keep
      create  test/integration
      create  test/integration/.keep
      create  test/channels/application_cable/connection_test.rb
      create  test/test_helper.rb
      create  test/system
      create  test/system/.keep
      create  test/application_system_test_case.rb
      remove  config/initializers/assets.rb
      remove  app/assets/config/manifest.js
      remove  app/assets/config
      remove  app/assets/stylesheets/application.css
      create  app/assets/stylesheets/application.css
      remove  app/models/application_record.rb
      remove  app/jobs
      remove  app/javascript/channels
      remove  app/channels
      remove  test/channels
      remove  config/initializers/cors.rb
      remove  config/initializers/new_framework_defaults_7_0.rb
         run  bundle install
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as root will break this application for all non-root users on this machine.
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
Fetching gem metadata from https://rubygems.org/...........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Using rake 13.0.6
Fetching concurrent-ruby 1.2.0
Installing concurrent-ruby 1.2.0
Using i18n 1.12.0
Fetching minitest 5.17.0
Installing minitest 5.17.0
Fetching tzinfo 2.0.6
Installing tzinfo 2.0.6
Fetching activesupport 7.0.4.2
Installing activesupport 7.0.4.2
Using builder 3.2.4
Fetching erubi 1.12.0
Installing erubi 1.12.0
Fetching mini_portile2 2.8.1
Installing mini_portile2 2.8.1
Fetching racc 1.6.2
Installing racc 1.6.2 with native extensions
Fetching nokogiri 1.14.1 (x86_64-linux)
Installing nokogiri 1.14.1 (x86_64-linux)
Using rails-dom-testing 2.0.3
Using crass 1.0.6
Fetching loofah 2.19.1
Installing loofah 2.19.1
Fetching rails-html-sanitizer 1.5.0
Installing rails-html-sanitizer 1.5.0
Fetching actionview 7.0.4.2
Installing actionview 7.0.4.2
Fetching rack 2.2.6.2
Installing rack 2.2.6.2
Using rack-test 2.0.2
Fetching actionpack 7.0.4.2
Installing actionpack 7.0.4.2
Using nio4r 2.5.8
Using websocket-extensions 0.1.5
Using websocket-driver 0.7.5
Fetching actioncable 7.0.4.2
Installing actioncable 7.0.4.2
Fetching globalid 1.1.0
Installing globalid 1.1.0
Fetching activejob 7.0.4.2
Installing activejob 7.0.4.2
Fetching activemodel 7.0.4.2
Installing activemodel 7.0.4.2
Fetching activerecord 7.0.4.2
Installing activerecord 7.0.4.2
Using marcel 1.0.2
Using mini_mime 1.1.2
Fetching activestorage 7.0.4.2
Installing activestorage 7.0.4.2
Fetching date 3.3.3
Installing date 3.3.3 with native extensions
Fetching timeout 0.3.1
Installing timeout 0.3.1
Fetching net-protocol 0.2.1
Installing net-protocol 0.2.1
Fetching net-imap 0.3.4
Installing net-imap 0.3.4
Using net-pop 0.1.2
Using net-smtp 0.3.3
Fetching mail 2.8.0.1
Installing mail 2.8.0.1
Fetching actionmailbox 7.0.4.2
Installing actionmailbox 7.0.4.2
Fetching actionmailer 7.0.4.2
Installing actionmailer 7.0.4.2
Fetching actiontext 7.0.4.2
Installing actiontext 7.0.4.2
Fetching public_suffix 5.0.1
Installing public_suffix 5.0.1
Using addressable 2.8.1
Using bindex 0.8.1
Using msgpack 1.6.0
Fetching bootsnap 1.16.0
Installing bootsnap 1.16.0 with native extensions
Using bundler 2.1.4
Using matrix 0.4.2
Fetching regexp_parser 2.6.2
Installing regexp_parser 2.6.2
Using xpath 3.2.0
Using capybara 3.38.0
Fetching io-console 0.6.0
Installing io-console 0.6.0 with native extensions
Fetching reline 0.3.2
Installing reline 0.3.2
Fetching irb 1.6.2
Installing irb 1.6.2
Fetching debug 1.7.1
Installing debug 1.7.1 with native extensions
Using method_source 1.0.0
Using thor 1.2.1
Using zeitwerk 2.6.6
Fetching railties 7.0.4.2
Installing railties 7.0.4.2
Fetching importmap-rails 1.1.5
Installing importmap-rails 1.1.5
Using jbuilder 2.11.5
Using puma 5.6.5
Fetching rails 7.0.4.2
Installing rails 7.0.4.2
Using rexml 3.2.5
Using rubyzip 2.3.2
Using websocket 1.2.9
Fetching selenium-webdriver 4.8.0
Installing selenium-webdriver 4.8.0
Using web-console 4.2.0
Using webdrivers 5.2.0
Bundle complete! 11 Gemfile dependencies, 68 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
         run  bundle binstubs bundler
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
       rails  importmap:install
rails aborted!
NoMethodError: undefined method `action_mailer' for #<Rails::Application::Configuration:0x0000000003cef3d0>
/rails/myApp/config/environments/development.rb:37:in `block in <main>'
/rails/myApp/config/environments/development.rb:3:in `<main>'
/rails/myApp/config/environment.rb:5:in `<main>'
Tasks: TOP => app:template => environment
(See full trace by running task with --trace)
```

### System configuration
**Rails version**: 7.0.4.2

**Ruby version**: 2.7.3

### Workaround
Uncommenting the missing require in config/application.rb allows starting Rails after the failed install, but it's hard to tell if anything important is missing.  Creating a new app with all the unwanted components works.

**Comment by skipkayhil** [comment]
Hey @ekortright, thanks for the really detailed issue! This should be fixed on `7-0-stable` in #46308, and will be included in `7.0.5` whenever that is released.

I'm going to close this, but feel free to reply if I've missed something!

**Comment by ekortright** [comment]
Hi @skipkayhil, thank you for the link to the fix.  Based on that, I just removed `--skip-active-job' from the `rails new` command and the app creation completed with no problems.
